### PR TITLE
Share the  OrchestratorClient imlementation of RecipientContextGenerator

### PR DESCRIPTION
**Pull request by jblebrun**
Currently, any trusted app implementation needs to create an instance of OrchestratorClient to provide to ServerEncryptor as a RecipientGeneratorContext.

There may be further refactors to remove direct dependency on OrchestratorClient completely, but this unblocks external trusted app development for now.

### CODE REVIEW

**Comment by tiziano88** [comment]
@jul-sh @ipetr0v could you review this? I am confused as to why this was not already the case. Also the readme says that `oak_containers_orchestrator_client` connects *from* the orchestrator to the launcher? That is not what I expected.

https://github.com/project-oak/oak/blob/9f22a5f03c7a141aa1b265b33efd2f2469fa6846/oak_containers_orchestrator_client/README.md?plain=1#L9-L10

**Comment by tiziano88** [comment]
at a closer look, `oak_containers_orchestrator_client` appears to contain the launcher client, not the orchestrator client. Not sure why -- is that intentional @ipetr0v @jul-sh ?

**Review by ipetr0v — COMMENTED** [review]
As part of implementing the Oak SDK for enclave applications, our current plan is to not use the `OrchestratorClient` for encryption. For this we will have a dedicated gRPC client implementation, which will be hidden behind a new `EncryptionKeyHandle` class. Which will connect to the new Orchestrator Crypto Service: https://github.com/project-oak/oak/blob/main/oak_containers_orchestrator/src/crypto.rs

Also, keep in mind that `oak_containers_orchestrator_client` is a gRPC client from the Orchestrator to the Hostlib. And at the same time we have an Orchestrator Client concept, which is a gRPC client from the Enclave app to the Orchestrator.

**Comment by tiziano88** [comment]
@ipetr0v @jul-sh so what do you recommend as next step about this PR?

**Comment by ipetr0v** [comment]
> @ipetr0v @jul-sh so what do you recommend as next step about this PR?

I'm currently adding a new `oak_containers_sdk` crate where I add a gRPC client for the Orchestrator Crypto service: https://github.com/project-oak/oak/pull/4602

Once this PR is submitted, the existing `OrchestratorClient` could be moved there. And the crypto logic would be removed from it, leaving only the `get_config` and `notify` functions. I.e. once the enclave application has started and notified the Orchestrator, the Orchestrator Client could be dropped.

File: oak_containers_orchestrator_client/src/lib.rs [hunk 1]
```diff
@@ -44,6 +44,8 @@ use self::proto::oak::{
     key_provisioning::v1::GroupKeys,
     session::v1::AttestationEvidence,
 };
+mod orchestrator_client;
+pub use crate::orchestrator_client::OrchestratorClient;
```
Comment (jul-sh) [inline oak_containers_orchestrator_client/src/lib.rs:48]: It looks like this crate does not use the OrchestratorClient, it only exposes it?

**Comment by jul-sh** [comment]
> > @ipetr0v @jul-sh so what do you recommend as next step about this PR?
> 
> I'm currently adding a new `oak_containers_sdk` crate where I add new a gRPC client for the Orchestrator Crypto service: #4602
> 
> Once this PR is submitted, the existing `OrchestratorClient` could be moved there. And the crypto logic would be removed from the orchestrator client, leaving only the `get_config` and `notify` functions. I.e. once the enclave application has started and notified the Orchestrator, the Orchestrator Client could be dropped.

Sounds good! I think we should move `oak_containers_hello_world_trusted_app/src/orchestrator_client.rs` into the SDK. That would accomplish the goal of this PR in the most idiomatic manner I think. 

To unblock external trusted app development in the meantime, I would recommend copying and pasting the relevant code contained in `oak_containers_hello_world_trusted_app/src/orchestrator_client.rs` into each trusted app. The trusted app can the migrate to the SDK later. 

@jblebrun does that sound feasible? I think the intention of the PR to share this logic is really good., thank you! It shouldn't go into (confusingly named) `oak_containers_orchestrator_client`, but instead into the SDK.

**Comment by jul-sh** [comment]
Separately I'll open a PR to remove the `oak_containers_orchestrator_client`, as it is confusingly named. Really I think that crate should just be a part of the orchestrator itself. It contains the launcher client, that's only used by the orchestrator.

**Comment by jblebrun** [comment]
> Separately I'll open a PR to remove the `oak_containers_orchestrator_client`, as it is confusingly named. Really I think that crate should just be a part of the orchestrator itself. It contains the launcher client, that's only used by the orchestrator.

I started a PR to do that: https://github.com/project-oak/oak/pull/4603 -- let me know if that's what you had in mind.

**Pull request closed without merging**